### ISSUE TITLE
✨ Add convenience function to easily slice data

### ISSUE THREAD

**Issue by s-weigand** [glotaran/pyglotaran#905]
### Is your feature request related to a problem? Please describe.

Sometimes you need to select data before the analysis or just want to reduce code duplication when plotting a specific region.
Currently, you need to operate directly on `xarray.Dataset`s and need to use `isel` and `masks`.

Example from [one of my notebooks](https://github.com/s-weigand/pyglotaran-examples/blob/notebooks/notebooks/ex_doas_beta/ex_doas_beta.ipynb):

```python
def plot_func_factory(dataset: xr.DataArray):
    time = dataset.time

    def wrapper(spectral: float, time_range: tuple[float, float], doas: list[str]):
        time_mask = np.logical_and(time >= time_range[0], time <= time_range[1])
        doas_mask = np.isin(dataset.damped_oscillation, doas)
        dataset.isel(time=time_mask, damped_oscillation=doas_mask).sel(
            spectral=spectral, method="nearest"
        ).plot.line(x="time")
        plt.show()

    return wrapper


plot_cos = plot_func_factory(result.data["dataset1"]["damped_oscillation_cos"])

```

### Describe the solution you'd like

An API something like this would be nice.

```python
from glotaran.utils.io import data_selector

selected_data = data_selector(
    dataset, 
    coord_selection={
        "time": (0, 1.5), 
        "damped_oscillation": ["doa_1", "doa_2", "doa_2"],
    }
)

selected_data.sel(spectral=spectral, method="nearest").plot.line(x="time")        
plt.show()
```

### Describe alternatives you've considered

We could also implement it in the extras, BUT @ism200  often said in meetings that to prepare input data he would need to use his legacy software which we are trying to replace. And since preparing data is the responsibility of a fitting engine IMHO here is the place such functionality belongs.

**Comment by joernweissenborn** [comment]
The wrapper could be  

```py
def plot_func_factory(dataset: xr.DataArray):

    def wrapper(spectral: float, time_range: tuple[float, float], doas: list[str]):
        dataset.sel(time=slice(time_range[0], time_range[1]).sel(damped_oscillation=doas).sel(
            spectral=spectral, method="nearest"
        ).plot.line(x="time")
        plt.show()

    return wrapper


plot_cos = plot_func_factory(result.data["dataset1"]["damped_oscillation_cos"])
```

maybe reorder sels for performance reasons.

**Comment by s-weigand** [comment]
Resolved in #1029
